### ISSUE TITLE
Shouldn't assume WebApplicationContext so Togglz can be used for non-web apps

### ISSUE THREAD

**Issue by rpomeroy** [togglz/togglz#55]
SpringBeanFinder assumes WebApplicationContext.  This doesn't work if using Togglz in a Spring command line app (which Springsource is providing a lot of support for).

**Comment by chkal** [comment]
@rpomeroy: Thanks for sharing these details. 

The Spring support is now divided into two modules. The first one `togglz-spring-core` only depends on core Spring APIs and `togglz-spring-web` adds more support on top for web environments.

So in your specific usecase you can now simply add the `togglz-spring-core` dependency to your project and then bootstrap Togglz like this:

```java
FeatureManager manager = new FeatureManagerBuilder()
        .featureClass(MyFeaturess.class)
        .stateRepository(new InMemoryStateRepository())
        .userProvider(new NoOpUserProvider())
        .build();

StaticFeatureManagerProvider.setFeatureManager(manager);
```

Using `StaticFeatureManagerProvider` is absolutely fine if you are running a Java SE application with a single classloader and one application per VM.

The only thing that is still hard to do is to let Spring manage the FeatureManager instance. But even this should be possible if you do something like after you started the application context:

```java
FeatureManager manager = applicationContext.getBean(FeatureManager.class);
StaticFeatureManagerProvider.setFeatureManager(manager);
```

But I'm still looking for a better solution for this...

**Comment by Iker-Jimenez** [comment]
Hi,
our setup is pretty much the same as the one described by @rpomeroy. What snapshot version of togglz has the new modules so we can test this out?

Thanks.

**Comment by chkal** [comment]
You can give `2.2.0-SNAPSHOT` a try. See:

https://oss.sonatype.org/content/repositories/snapshots/org/togglz/togglz-spring-core/2.2.0-SNAPSHOT/

All details about how to access the snapshot builds are available here:

http://www.togglz.org/download.html

**Comment by Iker-Jimenez** [comment]
OK, we have upgraded to 2.1.0 and played with 2.2.0-SNAPSHOT.
The StaticFeatureManagerProvider released in 2.1.0 was a big help, we were using our own BeanFinder before that.
2.2.0 adds a nice to have Spring compatible factory bean, so we would like to use it once you release 2.2.0.Final. Any dates for this release in mind?

**Comment by chkal** [comment]
Hey. Thanks for the feedback. Sounds like it is working find for you. ;)

I'm not sure about the release date for 2.2.0.Final yet. We just pushed out 2.1.0.Final about 1 week ago. So this will likely take some time. Perhaps I'll do a 2.2.0.Alpha|Beta in between.
